### PR TITLE
fix: selinux handling and apparmor tests

### PR DESCRIPTION
**Pull request by frezbo**
Conditionally mount selinuxfs only if it's present.

Fix AppArmor tests, `apparmor` and other minor LSM's and set
`apparmor=1`.

### CODE REVIEW

File: internal/pkg/mount/v2/pseudo.go [hunk 1]
```diff
@@ -43,5 +42,12 @@ func PseudoSubMountPoints() Points {
 		)
 	}
 
+	if _, err := os.Stat("/sys/fs/selinux"); err == nil {
+		// mount selinuxfs if it exists
```
Comment (dsseng) [inline internal/pkg/mount/v2/pseudo.go:46]: In which cases it's not present? When using AppArmor as the major LSM?

File: pkg/provision/providers/qemu/node.go [hunk 2]
```diff
@@ -84,9 +84,6 @@ func (p *provisioner) createNode(state *vm.State, clusterReq provision.ClusterRe
 	// Talos config
 	cmdline.Append("talos.platform", constants.PlatformMetal)
 
-	// SELinux
-	cmdline.Append("selinux", "1")
-
 	// add overrides
 	if nodeReq.ExtraKernelArgs != nil {
 		if err = cmdline.AppendAll(nodeReq.ExtraKernelArgs.Strings()); err != nil {
```
Comment (smira) [inline pkg/provision/providers/qemu/node.go:89]: can we instead use negative merging here and pass `-selinux` in case of apparmor test?

Comment (smira) [reply pkg/provision/providers/qemu/node.go:89]: I would rather leave `selinux=1` enabled by default for now

File: internal/pkg/mount/v2/pseudo.go (hunk 1 above)
Comment (frezbo) [reply internal/pkg/mount/v2/pseudo.go:46]: yes, when lsm  is apparmor

File: pkg/provision/providers/qemu/node.go (hunk 2 above)
Comment (frezbo) [reply pkg/provision/providers/qemu/node.go:89]: so check if we pass `apparmor=1`, then add `-selinux`?

Comment (frezbo) [reply pkg/provision/providers/qemu/node.go:89]: actually I got it, better way

Comment (smira) [reply pkg/provision/providers/qemu/node.go]: I mean in the `e2e-qemu.sh` directly if apparmor was requrest

Comment (frezbo) [reply pkg/provision/providers/qemu/node.go]: yeh, did the same

**Review by dsseng — APPROVED** [review]

**Pull request merged**